src/Language/PureScript/Parser/Declarations.hs

```diff
@@ -492,6 +492,25 @@ parseNumberBinder = NumberBinder <$> (sign <*> number)
          <|> (symbol' "+" >> return id)
          <|> return id
 
+parseTypedBinder :: TokenParser Binder
+parseTypedBinder = buildPostfixParser postfixTable typeableBinder P.<?> "typed binder"
+    where
+      postfixTable = [ \b -> flip TypedBinder b <$> (P.try (indented *> doubleColon) *> parseType)
+                     ]
+      typeableBinder = P.choice (map P.try
```

Comment:
Awesome, I think I've got it now? That definitely removed some duplication that was bugging me.